app/vmalert/web_types.go

```diff
@@ -151,13 +154,24 @@ type apiRule struct {
 	ID string `json:"id"`
 	// GroupID is an unique Group's ID
 	GroupID string `json:"group_id"`
+	// GroupName is Group name rule belong to
+	GroupName string `json:"group_name"`
+	// File is file name where rule is defined
+	File string `json:"file"`
 	// Debug shows whether debug mode is enabled
 	Debug bool `json:"debug"`
 
 	// MaxUpdates is the max number of recorded ruleStateEntry objects
 	MaxUpdates int `json:"max_updates_entries"`
 	// Updates contains the ordered list of recorded ruleStateEntry objects
-	Updates []rule.StateEntry `json:"-"`
+	Updates []rule.StateEntry `json:"updates,omitempty"`
+}
+
+func (ar apiRule) toJSON(withField string) ([]byte, error) {
```

Comment:
those method won't be invoked for json.Marshal for apiGroup, which includes apiRule. And marshaling will be incorrect.

Reply:
I updated the logic. Can you take a look, please?